x-pack/plugins/rollup/server/lib/search_strategies/rollup_search_capabilities.js

```diff
@@ -0,0 +1,40 @@
+/*
+* Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+* or more contributor license agreements. Licensed under the Elastic License;
+* you may not use this file except in compliance with the Elastic License.
+*/
+import { get } from 'lodash';
+
+const intervalMultiple = (userTimeInterval, defaultTimeInterval) => !Boolean(userTimeInterval % defaultTimeInterval);
+
+export default (DefaultSearchCapabilities) =>
+  (class RollupSearchCapabilities extends DefaultSearchCapabilities {
+    constructor(req, batchRequestsSupport, fieldsCapabilities) {
+      super(req, batchRequestsSupport, fieldsCapabilities);
+
+      this.init();
+    }
+
+    get fixedTimeZone() {
+      return get(this.dateHistogram, 'time_zone', null);
+    }
+
+    get defaultTimeInterval() {
+      return get(this.dateHistogram, 'interval', null);
+    }
+
+    init() {
+      this.dateHistogram = this.getDateHistogramAggregation();
+
+      this.validateTimeIntervalRules = [
+        intervalMultiple,
+      ];
+    }
+
+    getDateHistogramAggregation() {
+      const dateHistogramFields = this.getFieldsByAggregationType('date_histogram');
+      const keys = Object.keys(dateHistogramFields);
```

Comment:
Could you explain what that's supposed to do? This looks a bit like we're implicitly relying on the order that `Object.keys` are returning the keys, which is NOT guaranteed to be in any specific order.

Reply:
Agree with you, it's not obvious... I'll fix to make more readable. 

Let me explain: 
getFieldsByAggregationType method returns all fields which supports passed aggregation in the following format:

```
{
   field_name1: {
        agg: 'histogram',
        option1: 'option1', 
        option2: 'option2' ,
   },
    field_name2: {
        agg: 'histogram',
        option1: 'option1', 
        option2: 'option2',
   }
}
```
Rollup Job can have only one 'date_histogram'. So I take the first value from that object.

Reply:
Fixed